### PR TITLE
Re-organize algorithms so TOC matches README

### LINKED ISSUE #24347: [docs][rllib] Reorganize the RLlib docs Algorithms section
### Description

Proposed Re-grouping of Algorithms in Docs

1. Rename UCB -> LinUCB on the [algorithms doc page](https://docs.ray.io/en/master/rllib/rllib-algorithms.html#available-algorithms-overview)
2. Add a friendly list of algorithms on the github RLlib page itself's [README](https://github.com/ray-project/ray/tree/master/rllib)
3. Repeat same groupings on the [algorithms doc page](https://docs.ray.io/en/master/rllib/rllib-algorithms.html#available-algorithms-overview) as the [README](https://github.com/ray-project/ray/tree/master/rllib)
4. Add links that go from README listed algorithms to algorithms doc page

README Algo organization 
Offline RL (put Off-policy Eval here):  BC, DQN, DDPG, TD3, SAC, CQL,
Model-free RL (maybe split into off/on policy)
Model-based RL
Derivative-free Algos
RecSys and Bandits (or separate categories)
Games (especially AlphaZero, APPO, IMPALA, A2C, A3C goes here)

### Use case

People come to RLlib looking for combinations of Environment and Algorithms that might best work for their Use Case.  The goal of the re-org is to give people more guidance finding which algorithms might be worth a try.

The algorithms today are listed in docs in API format.  They are not listed at all on the github README.md for RLlib itself.

### PR DESCRIPTION
**Pull request by christy**
Algorithms page Table of Contents (TOC) need to match new algorithms grouping on [RLlib github README](https://github.com/ray-project/ray/blob/master/rllib/README.rst) page.

## Why are these changes needed?

Algorithms were re-ordered on the page so the order would show up in TOC in same order as github README.

## Related issue number

Closes #24347
Duplicates #25823

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [x] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/rllib/rllib-algorithms.rst [hunk 1]
```diff
@@ -75,105 +75,98 @@ Algorithm                        Frameworks Discrete Actions        Continuous A
...
-**IMPALA-specific configs** (see also `common configs <rllib-training.html#common-parameters>`__):
+**MARWIL-specific configs** (see also `common configs <rllib-training.html#common-parameters>`__):
 
-.. literalinclude:: ../../../rllib/algorithms/impala/impala.py
+.. literalinclude:: ../../../rllib/algorithms/marwil/marwil.py
    :language: python
    :start-after: __sphinx_doc_begin__
    :end-before: __sphinx_doc_end__
+   
+Model-free On-policy RL (for Games)
```
[the remark this answers is not in the document]
Comment (christy) [reply doc/source/rllib/rllib-algorithms.rst:168]: Done, removed.

**Comment by christy** [comment]
Merged the nits fixes.  1 test about tests logging failed, but it was nothing to do with my code.

**Comment by gjoliver** [comment]
we intentionally want to list Offline methods in front of Online methods right?
just to double check that it is a deliberate decision.

**Review by sven1977 — APPROVED** [review]
LGTM, let's merge.
I think it's ok to list offline first. It's a distinctive feature of RLlib that we support so many offline RL algos. Fine with me.

**Pull request merged**